ydb/core/tx/datashard/prefix_kmeans.cpp

```diff
@@ -494,28 +470,100 @@ class TPrefixKMeansScan final: public TPrefixKMeansScanBase, private TCalculatio
             cells[0] = TCell::Make(Parent);
             TSerializedCellVec::UnsafeAppendCells(cells, pk);
             PrefixBuf.AddRow(TSerializedCellVec{std::move(pk)}, TSerializedCellVec::Serialize({}));
-        } else if (!TCellVectorsEquals{}(Prefix.GetCells(), key.subspan(0, PrefixColumns))) {
-            HasNextPrefix = true;
-            return EScan::Reset;
         }
-        ++ReadRows;
-        ReadBytes += CountBytes(key, row);
-        switch (State) {
-            case EState::SAMPLE:
-                return FeedSample(row);
-            case EState::KMEANS:
-                return FeedKMeans(row);
-            case EState::UPLOAD_BUILD_TO_BUILD:
-                return FeedUploadBuild2Build(key, row);
-            case EState::UPLOAD_BUILD_TO_POSTING:
-                return FeedUploadBuild2Posting(key, row);
-            default:
-                Y_ASSERT(false);
-                return EScan::Final;
+
+        if (IsFirstPrefixFeed && IsPrefixRowsValid) {
+            PrefixRows.AddRow(TSerializedCellVec{key}, TSerializedCellVec::Serialize(*row));
+            if (PrefixRows.IsReachLimits(Limits)) {
+                PrefixRows.Clear();
+                IsPrefixRowsValid = false;
+            }
```

Comment:
Doesn't really matter, but I think it's better
```suggestion
            if (PrefixRows.IsReachLimits(Limits)) {
                PrefixRows.Clear();
                IsPrefixRowsValid = false;
            } else {
                PrefixRows.AddRow(TSerializedCellVec{key}, TSerializedCellVec::Serialize(*row));
            }
```

Reply:
This way we check limit as soon as possible without waiting next iteration with broken limits, and that seems more sensible

But doesn't matter so let's leave as is